sparse-file.go

```diff
@@ -214,30 +223,21 @@ func (l *sparseFileLoader) loadChunk(i int) error {
 		}
 
 		l.mu.Lock()
-		l.done[i] = true
+		l.done.Set(i, true)
 		l.mu.Unlock()
 	})
 	return loadErr
 }
 
 // writeState saves the current internal state about which chunks have
-// been loaded. It's basically just a sequence of '0' and '1' of the
+// been loaded. It's a bitmap of the
 // same length as the index, with 0 = chunk has not been loaded and
-// 1 = chunk has been loaded. TODO: could do that with a proper bitmap
-// to save space in the file.
-func (l *sparseFileLoader) writeState(w io.Writer) error {
+// 1 = chunk has been loaded.
+func (l *sparseFileLoader) writeState() error {
 	l.mu.Lock()
 	defer l.mu.Unlock()
-	b := make([]byte, 0, len(l.done))
-	for _, done := range l.done {
-		if done {
-			b = append(b, '1')
-		} else {
-			b = append(b, '0')
-		}
-	}
-	_, err := w.Write(b)
-	return err
+
+	return ioutil.WriteFile(l.name+".state", l.done.Data(false), 0755)
```

Comment:
Sorry, one more, I think this should be be 0644 since it's not executable

Reply:
ok ! sorry and thanks a lot
copy/paste is not our friend :)